packages/babel-core/src/parse.js

```diff
@@ -0,0 +1,22 @@
+// @flow
+
+import loadConfig, { InputOptions } from "./config";
+import normalizeFile from "./transformation/normalize-file";
+import normalizeOptions from "./transformation/normalize-opts";
+
+type AstRoot = BabelNodeFile | BabelNodeProgram;
+
+export default function parse(
+  code: string,
+  opts?: InputOptions,
+): AstRoot | null {
+  const config = loadConfig(opts);
+
+  if (config === null) {
```

Comment:
This was the main question I had when thinking through this originally. I wonder if it might be better do do `config || {}` so if a file was ignore, it is just like no config was given, instead of returning `null` since that's not that useful from what I can see. Tempted to do the same for the rest of the Babel functions.

Reply:
So would this then try to parse without a config, potentially throwing errors if it encounters syntax that it needs a plugin for?

Reply:
Yeah IDK, I've been thinking about this more and I think I'm changing back to this being fine as-is. If Babel returns `null` them maybe ESLint can just use its default parser and we go from there?

Reply:
We can figure that out on the babel-eslint side. If this is the desired API for Babel, I think babel-eslint has a few options:

- Throw an error and show a helpful warning about needing a .babelrc or needing to define parserOptions in the ESLint config or to be passed on the command line.
- Fallback to parsing with Babylon with the current [default list of plugins](https://github.com/babel/babel-eslint/blob/master/lib/parse.js#L18-L45) (may not be up to date). This would make the fallback be essentially the behavior it has now.
- Fallback to trying to parse with Espree (ESLint's default parser). Not sure how I feel about this, because a lot of things might work/not work by accident, and I think this will end up being really confusing for users.

Reply:
yeah 1 or 3 makes sense. Not sure what case you would use it without a config (I guess you might if you have a preset and use the cli directly like `babel src -d lib --preset your-preset`), so maybe 3 but then ESLint wouldn't pick up any syntax lol.

Reply:
I'm initially leaning towards number 1